addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxServerHandler.java

```diff
@@ -647,74 +659,63 @@ public void updateListener(SqueezeBoxPlayerEventListener listener) {
                 // Parameter Genre
                 else if (messagePart.startsWith("genre%3A")) {
-                    final String value = messagePart.substring("genre%3A".length());
-                    updatePlayer(new PlayerUpdateEvent() {
-                        @Override
-                        public void updateListener(SqueezeBoxPlayerEventListener listener) {
-                            listener.genreChangeEvent(mac, decode(value));
-                        }
-                    });
+                    genre = messagePart.substring("genre%3A".length());
                 }
                 // Parameter Year
                 else if (messagePart.startsWith("year%3A")) {
-                    final String value = messagePart.substring("year%3A".length());
-                    updatePlayer(new PlayerUpdateEvent() {
-                        @Override
-                        public void updateListener(SqueezeBoxPlayerEventListener listener) {
-                            listener.yearChangeEvent(mac, decode(value));
-                        }
-                    });
+                    year = messagePart.substring("year%3A".length());
                 }
-                // Parameter Artwork
-                else if (messagePart.startsWith("artwork_track_id%3A")) {
-                    String url = messagePart.substring("artwork_track_id%3A".length());
-                    // NOTE: what is returned if not an artwork id? i.e. if a
-                    // space?
-                    if (!url.startsWith(" ")) {
-                        url = "http://" + host + ":" + webport + "/music/" + url + "/cover.jpg";
+                // Parameter Artwork_url
+                else if (messagePart.startsWith("artwork_url%3A")
+                        && !messagePart.startsWith("artwork_url%3Ahttp%3A%2F%2F")) {
+                    url = messagePart.substring("artwork_url%3A".length());
+                    if (url.startsWith("%2F")) {
+                        url = "http://" + host + ":" + webport + decode(url);
+                    } else {
+                        url = "http://" + host + ":" + webport + "/" + decode(url);
                     }
-                    final String value = url;
-                    updatePlayer(new PlayerUpdateEvent() {
-                        @Override
-                        public void updateListener(SqueezeBoxPlayerEventListener listener) {
-                            listener.coverArtChangeEvent(mac, decode(value));
-                        }
-                    });
                 } else {
                     // Added to be able to see additional status message types
                     logger.trace("Unhandled status message type '{}'", messagePart);
                 }
             }
+            // update cover art
+            // update remote title
+            // update artist
+            // update album
+            // update genre
+            // update year
+
+            final String f_url = url;
```

Comment:
Please leave the `final` and `f_url` is not an accepted name according to the guidelines of (Oracle and) us.

Reply:
`final String  finalUrl = url`
is it acceptable ?

Reply:
Is it still technically required in java8, to have these additional finals

Reply:
Why did you group them in one call, for efficiency?

Reply:
yes

Reply:
Sorry I asked you two questions, both yes or only one?

> Is it still technically required in java8, to have these additional finals

> Why did you group them in one call, for efficiency?